[MRG+1] Spellchecked 'working with text data' tutorial

tommyod opened the pull request:
#### What does this implement/fix? Explain your changes.

This is a small PR, which introduces minor changes to the tutorial on working with text data:

- Rephrased a couple of sentences.
- Added a few links to classes used in the tutorial.
- Added some punctuation.

Comments very welcome.

tommyod committed:
Spellchecked 'working with text data' tutorial

jnothman commented on doc/tutorial/text_analytics/working_with_text_data.rst:
```diff
@@ -20,22 +20,24 @@ In this section we will see how to:
 
-The source of this tutorial can be found within your
-scikit-learn folder::
+The source of this tutorial can be found within your scikit-learn installation,
```
it's not really an installation...

jnothman commented on doc/tutorial/text_analytics/working_with_text_data.rst:
```diff
@@ -172,26 +174,26 @@ turn the text content into numerical feature vectors.
 larger than 100,000.
 
-If ``n_samples == 10000``, storing ``X`` as a numpy array of type
+If ``n_samples = 10000``, storing ``X`` as a NumPy array of type
```
I think this is clearer with ==

jnothman commented on doc/tutorial/text_analytics/working_with_text_data.rst:
```diff
@@ -172,26 +174,26 @@ turn the text content into numerical feature vectors.
 
 Fortunately, **most values in X will be zeros** since for a given
-document less than a couple thousands of distinct words will be
+document less than a couple thousand of distinct words will be
```
let's say "a few thousand distinct"

jnothman reviewed (commented):
I suppose this is generally an improvement. Yes, some of us use british spellings.

qinhanmin2014 commented on doc/tutorial/text_analytics/working_with_text_data.rst:
```diff
@@ -311,7 +316,7 @@ Building a pipeline
 
 In order to make the vectorizer => transformer => classifier easier
-to work with, ``scikit-learn`` provides a ``Pipeline`` class that behaves
+to work with, ``scikit-learn`` provides a :class:`Pipeline` class that behaves
```
this won't work, You should either change currentmodule or try sklearn.pipeline.Pipeline

qinhanmin2014 reviewed (approved):
LGTM overall

tommyod committed:
Addressed reviewer comments

tommyod commented:
@jnothman @qinhanmin2014 Thanks for the review! I've addressed your comments. It's a small PR, but I'm trying to get a good feel for git/github/sklearn - gotta start somewhere. Let me know if there's anything else.

rth commented:
> Let me know if there's anything else.

Well there are  ["first good issue"](https://github.com/scikit-learn/scikit-learn/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) and ["help wanted"](https://github.com/scikit-learn/scikit-learn/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22") issue tags. It also depends on what part of the code you are interested in contributing to..

rth commented:
I meant, if you were asking about other issues contribute to, not other comments about this PR :)

qinhanmin2014 reviewed (approved):
LGTM

jnothman commented:
Try to avoid rebasing as it makes tracking changes harder. Thanks
